jax/experimental/jax2tf/tests/shape_poly_test.py

```diff
@@ -331,66 +332,258 @@ def test_stride_shape(self):
+      args = self.dyn_args_maker(tst.rng())
+
+      assert self.polymorphic_shapes is None
+      assert self.input_signature is None
+      assert args is not None and len(args) == len(poly_axes)
+      # Make the polymorphic_shapes and input_signature
+      polymorphic_shapes = []
+      input_signature = []
+      for arg, poly_axis in zip(args, poly_axes):
+        if poly_axis is None:
+          polymorphic_shapes.append(None)
+          input_signature.append(tf.TensorSpec(np.shape(arg), arg.dtype))
+        else:
+          def make_arg_polymorphic_shapes(poly_axis: Sequence[int]) -> Tuple[str, tf.TensorSpec]:
+            idx = -1
+            dims = []
+            tensorspec_dims: List[Optional[int]] = []
+            for i, d in enumerate(arg.shape):
+              if i in poly_axis:
+                idx += 1
+                dims.append(f"b{idx}")
+                tensorspec_dims.append(None)
+              else:
+                dims.append(str(d))
+                tensorspec_dims.append(d)
+            return ", ".join(dims), tf.TensorSpec(tensorspec_dims, arg.dtype)
+
+          arg_polymorphic_shapes, arg_tensorspec = make_arg_polymorphic_shapes(poly_axis)
+          polymorphic_shapes.append(arg_polymorphic_shapes)
+          input_signature.append(arg_tensorspec)
+
+    expect_error_type, expect_error_regex = self.expect_error
+    if self.skip_jax_run and self.arg_descriptors == ():
+      f_jax = self.fun
+    else:
+      f_jax = self.dyn_fun
+    if expect_error_type is not None:
+      with tst.assertRaisesRegex(expect_error_type, expect_error_regex):
+        f_tf = jax2tf.convert(f_jax, polymorphic_shapes=polymorphic_shapes,
+                              enable_xla=self.enable_xla)
+        f_tf_func = tf.function(
+            f_tf, autograph=False, input_signature=input_signature)
+        # Create tf.ConcreteFunction and check inferred output signature
+        f_tf_func.get_concrete_function(*input_signature)
+
+      return
+    else:
+      f_tf = jax2tf.convert(f_jax, polymorphic_shapes=polymorphic_shapes,
+                            enable_xla=self.enable_xla)
+      f_tf_func = tf.function(
+          f_tf, autograph=False, input_signature=input_signature)
+      # Create tf.ConcreteFunction and check inferred output signature
+      concrete_f_tf = f_tf_func.get_concrete_function(*input_signature)
+
+    if self.expected_output_signature:
+      # Strangely, output_shapes can be a single shape for a function with a
+      # single result, or a list/tuple of shapes.
+      concrete_output_tf_shape = concrete_f_tf.output_shapes
+      if not isinstance(concrete_output_tf_shape, (tuple, list)):  # Single result
+        assert not isinstance(self.expected_output_signature, (tuple, list))
+        expected_output_signature = [self.expected_output_signature]
+        concrete_output_tf_shape = [concrete_output_tf_shape]
+      else:
+        expected_output_signature = self.expected_output_signature
```

Comment:
This might be simpler?

```suggestion
      concrete_output_tf_shape = concrete_f_tf.output_shapes
      expected_output_signature = self.expected_output_signature
      if not isinstance(concrete_output_tf_shape, (tuple, list)):  # Single result
        assert not isinstance(expected_output_signature, (tuple, list))
        expected_output_signature = [expected_output_signature]
        concrete_output_tf_shape = [concrete_output_tf_shape]
```